src/toolbox.js

```diff
@@ -212,6 +212,20 @@ var Toolbox = GObject.registerClass({
         this._infoLabel.label = text;
     }
 
+    setNeedsAttention(val, source) {
+        let topicRow = this._topicsList.get_selected_row();
```

Comment:
Instead of getting the selected row and special-casing it, if I am reading the code correctly, you have the ID of the topic in `source` so you could probably use `_findTopic(source)` to get the row and the child widget regardless of whether it is selected or not.

Reply:
> Looks good, and I tested it out.
> 
> In addition to the comments on specific lines of code I have two general implementation suggestions that I think can make it easier to adapt to all the other toolboxes as well.
> 
> First, I would suggest making the `needs-attention` signal into a GObject property instead of a signal. A signal allows you to observe state changes, as you are doing here, but the property will allow that too (via the `notify::needs-attention` signal) as well as get/set the state directly. You could take the same property spec and getter/setter functions as used in `TopicButton`, and then instead of `_onNeedsAttention()` you could set the property.
> 
Are you suggesting that I add the same property that exists in the TopicButton class to the userFunction and controlPanel classes? The aspect of this that was difficult for me to wrap my head around was how I could access the property from the error checking code. I wasn't sure if I could access the property unless I had an instantiated TopicButton class in the class. 
> That would also allow the second improvement that I'm thinking of, which would be to automatically bind the topic panel's needsAttention state to the topic button's needsAttention indicator state. In `Toolbox.addTopic()` you could put a `widget.bind_property('needs-attention', topic, 'needs-attention', ...)` and avoid having to connect it by hand for every topic.
> 
> This would require adding the property to all the other widgets passed to `addTopic()` in the other toolboxes, but you could also make it optional for the time being by checking if the `needs-attention` property exists before trying to bind it. Ultimately we do want the error UX improvements to apply to all the other toolboxes as well, so the more of it that you can implement in `src/toolbox.js` the better.

Reply:
Exactly, I was suggesting to add the same property to the UserFunction and ControlPanel classes.

It could be completely independent of the TopicButton instance. The only place where they would come into contact would be in `Toolbox.addTopic()` which would bind the TopicButton's property to the UserFunction or ControlPanel's property.